hs_core/views/__init__.py

```diff
@@ -539,61 +637,157 @@ def create_resource_select_resource_type(request, *args, **kwargs):
+
+    add_language = True
+    for element in metadata:
+        if 'language' in element:
+            if 'code' in element['language']:
+                #language_code = element['language']['code']
+                add_language = False
+            else:
+                metadata.remove(element)
+            break
+
+    if add_language:
+        metadata.append({'language': {'code': 'eng'}})
+
+    # add the default rights/license element
+    metadata.append({'rights':
+                         {'statement': 'This resource is shared under the Creative Commons Attribution CC BY.',
+                          'url': 'http://creativecommons.org/licenses/by/4.0/'
+                         }
+                    })
 
     if url_key in page_url_dict:
         return render(request, page_url_dict[url_key], {'title': res_title, 'metadata': metadata})
 
-    try:
-        resource = hydroshare.create_resource(
-                resource_type=request.POST['resource-type'],
-                owner=request.user,
-                title=res_title,
-                keywords=None,
-                metadata=metadata,
-                files=request.FILES.getlist('files'),
-                content=res_title
-        )
-    except Exception as ex:
-        context = {'resource_creation_error': ex.message }
+    resource = hydroshare.create_resource(
+            resource_type=request.POST['resource-type'],
+            owner=request.user,
+            title=res_title,
+            keywords=None,
+            dublin_metadata=None,
+            metadata=metadata,
+            files=request.FILES.getlist('files'),
+            content=res_title
+    )
+
+    # Send post-create resource signal
+    post_create_resource.send(sender=res_cls, resource=resource, metadata=metadata, **kwargs)
+
+    if resource is not None:
+        # go to resource landing page
+        return HttpResponseRedirect(resource.get_absolute_url())
+    else:
+        context = {
+            'resource_creation_error': 'Resource creation failed'
+        }
         return render_to_response('pages/create-resource.html', context, context_instance=RequestContext(request))
 
-    try:
-        utils.resource_post_create_actions(resource=resource, user=request.user, metadata=metadata, **kwargs)
-    except (utils.ResourceFileValidationException, Exception) as ex:
-        request.session['file_validation_error'] = ex.message
 
-    # go to resource landing page
-    return HttpResponseRedirect(resource.get_absolute_url())
+class CreateResourceForm(forms.Form):
+    title = forms.CharField(required=True)
+    creators = forms.CharField(required=False, min_length=0)
+    contributors = forms.CharField(required=False, min_length=0)
+    abstract = forms.CharField(required=False, min_length=0)
+    keywords = forms.CharField(required=False, min_length=0)
+
+@login_required
+def create_resource(request, *args, **kwargs):
+    global resource
+    qrylst = request.POST
```

Comment:
i think we can afford the extra characters to make this variable name pronounceable.